textx/scoping/providers.py

```diff
@@ -524,6 +574,28 @@ def __init__(self, filename_pattern=None, glob_args=None):
             self, PlainName(), filename_pattern, glob_args=glob_args)
 
 
+def _objects_list_to_tuples_of_name_and_objects(
+        obj_list, name_resolver_logic, name_part, cls):
+    from textx import textx_isinstance
+    name_obj_tuples = list(map(
+        lambda x: (name_resolver_logic(x), x),
+        obj_list))
+
+    name_obj_tuples = list(filter(
+        lambda x: textx_isinstance(x[1], cls) and (
+                type(x[0]) is Postponed or
+                x[0].find(name_part) >= 0),
+        name_obj_tuples))
+
+    return name_obj_tuples
+
+
+def _filter_postponed_from_tuples_of_name_and_objects(
+        name_obj_tuples):
+    return list(filter(lambda x: type(x[0]) is not Postponed,
+                       name_obj_tuples))
```

Comment:
Same here. Is converting to list necessary? Lazy iterables are preferred for performance reasons.

Reply:
(No action, see comment below)